From 313854b285aa9857e82ea09017ebc95e2034f164 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Thu, 29 Sep 2022 18:57:41 +0530
Subject: [PATCH] ath12k: remove reserve member of wbm completion structure

In tx completion, status desc obtained from offsetting the address from wbm ring.
Having reserved 8 bytes in status desc offsets the address twice and read the
values from the incorrect address. So, removing the reserved 8 bytes from  wbm
completion structure.

Fixes: I42879fea("ath12k: update htt_tx_wbm_completion descriptor to version 3")

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp.h b/drivers/net/wireless/ath/ath12k/dp.h
index 9ddbafa..bb1ae30 100644
--- a/drivers/net/wireless/ath/ath12k/dp.h
+++ b/drivers/net/wireless/ath/ath12k/dp.h
@@ -420,7 +420,6 @@ struct ath12k_dp {
 #define HTT_TX_WBM_COMP_INFO2_ACK_RSSI		GENMASK(31, 24)
 
 struct htt_tx_wbm_completion {
-	u32 rsvd0[2];
 	u32 info0;
 	u32 info1;
 	u32 info2;
-- 
2.17.1

